### PR TITLE
(1/5) Rearrange software docs: Using renv

### PR DESCRIPTION
**Pull request by sjspielman**
Towards #458 

This is the first of several ("pseudostacked"!) PRs to break up the current software page. Note the updated scheme for this plan in https://github.com/AlexsLemonade/OpenScPCA-analysis/issues/458#issuecomment-2150560957

This PR adds the following:
- Initializes a reproducibility directory (just a stub index for now)
- Initializes a managing software directory with an `index.md`. This file is wholly new.
- Ports `renv` software content into a new file `using-renv.md` in the managing software directory. There are a few rephrasings here that I noticed, but no "big" changes besides the occasional updated wording.

----

Here's what you can expect for the additional PRs to close out this issue:
- A PR to add a `using-conda.md` doc
- A PR to add a `reporting-dependencies.md` doc
- [EDIT: this step added] A PR to add the primer page for #498 (since this will affect relative links 👇 )
- After all of the above (including this PR) go in, I'll file a PR which removes the original software dependencies doc, updates the `mkdocs.yml` file to match the current set of software files, and fixes all the relative links that break after the `mkdocs.yml` change 🎢  
  - I will likely also add some relative links to the `documenting-your-analysis.md` page that points to a lot of these changes

### CODE REVIEW

File: docs/ensuring-repro/managing-software/index.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+As different versions of the same software often have different behaviors/usage and can even produce different results, pinning versions ensures that results are much more reproducibile.
+- Environments make it easier for other researchers to run a module.
+Analysis modules will have files specifying the specific software dependencies and versions which researchers can use to re-create the environment to reproducibility run the module's code.
+
+We specifically recommend these two package managers to specify module-specific software environments:
+
+- The R package manager [`renv`](https://rstudio.github.io/renv/) for modules containing R code
+  - Before working with `renv`, we encourage you to read [the excellent introduction](https://rstudio.github.io/renv/articles/renv.html) for more information.
+- The package manager [conda](https://docs.conda.io/en/latest/), along with [`conda-lock`](https://conda.github.io/conda-lock/), for modules containing Python code.
+conda can also be used to manage standalone software packages that do not depend on a specific language.
```
Comment (jaclyn-taroni) [inline docs/ensuring-repro/managing-software/index.md:16]: I _believe_ this is their preferred style:

```suggestion
Conda can also be used to manage standalone software packages that do not depend on a specific language.
```

File: docs/ensuring-repro/managing-software/index.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Managing software environments
+
+A key way the OpenScPCA project ensures reproducibility is through the use of module-specific software environments.
+Using software environments has several benefits:
+
+- Software dependencies, including both packages and stand-alone command line tools, can be pinned to specific versions.
+As different versions of the same software often have different behaviors/usage and can even produce different results, pinning versions ensures that results are much more reproducibile.
```
Comment (jaclyn-taroni) [inline docs/ensuring-repro/managing-software/index.md:7]:
```suggestion
As different versions of the same software often have different behaviors/usage and can produce different results, pinning versions ensures that results are much more reproducible.
```

File: docs/ensuring-repro/managing-software/index.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+- Environments make it easier for other researchers to run a module.
+Analysis modules will have files specifying the specific software dependencies and versions which researchers can use to re-create the environment to reproducibility run the module's code.
+
+We specifically recommend these two package managers to specify module-specific software environments:
+
+- The R package manager [`renv`](https://rstudio.github.io/renv/) for modules containing R code
```
Comment (jaclyn-taroni) [inline docs/ensuring-repro/managing-software/index.md:13]:
```suggestion
- The R package manager [`renv`](https://rstudio.github.io/renv/) for modules containing R code.
```

**Comment by sjspielman** [comment]
> Just a few style comments, one of which is a type that would have been caught with running on spellcheck on the branch... so I'm wondering if setting that up is something we want to do?

Setting u is currently hanging out in #503, but I can add that change in here too. Can't hurt, it's a small enough change to have "in parallel"!

**Pull request merged**